pageserver/src/layered_repository/storage_layer.rs

```diff
@@ -73,13 +73,24 @@ pub enum PageVersion {
 ///
 /// Data needed to reconstruct a page version
```

Comment:
I removed the first sentence that's no longer true, and pushed.

For later:

Hmm, perhaps the return value of 'get_page_reconstruct_data' should be part of the struct. So it would be like this:

```
pub enum PageReconstructState {
    /// This layer didn't contain all the required data, the caller should look up
    /// the predecessor layer at the returned LSN and collect more data from there.
    Continue(Lsn),
    /// Got all the data needed to reconstruct the requested page
    Complete,
    /// This layer didn't contain data needed to reconstruct the page version at
    /// the returned LSN. This is usually considered an error, but might be OK
    /// in some circumstances.
    Missing(Lsn),
}
// or maybe rename to PageBuilder
pub struct PageReconstructData {
    pub state: PageReconstructState,
    pub records: Vec<(Lsn, ZenithWalRecord)>,
    pub page_img: Option<(Lsn, Bytes)>,
}
```
This way, it's clear from the struct itself what the current state of the operation is. The caller would first initialize it in the 'Continue' state (or perhaps there should be another Start state), and fill in a cached page_img if one is available. Each call to 'get_page_reconstruct_data' would add any records or page_img, and update the state.